src/main/java/com/pashonokk/dvdrental/controller/FilmController.java

```diff
@@ -0,0 +1,70 @@
+package com.pashonokk.dvdrental.controller;
+
+import com.pashonokk.dvdrental.dto.CategoryDto;
+import com.pashonokk.dvdrental.dto.FilmDto;
+import com.pashonokk.dvdrental.dto.FilmSavingDto;
+import com.pashonokk.dvdrental.dto.PageDto;
+import com.pashonokk.dvdrental.exception.BigSizeException;
+import com.pashonokk.dvdrental.service.FilmService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.util.List;
+
+@Controller
+@RequiredArgsConstructor
+@RequestMapping("/films")
+public class FilmController {
+    private final FilmService filmService;
+
+    @GetMapping
+    public ResponseEntity<Page<FilmDto>> getFilms(@RequestBody PageDto pageDto) {
```

Comment:
Using @RequestParam instead of @RequestBody for GET requests: Typically, we don't send bodies in GET requests, and many HTTP clients do not support this. The parameters for page, size, and sort should probably be @RequestParams. Also, be sure to set some default values in case the client does not provide them.